asteroidsyncserviced/dbusinterface.h

```diff
@@ -59,23 +61,26 @@ public slots:
     bool StatusScreenshotService();
     bool StatusWeatherService();
     void RequestScreenshot();
-    void WeatherSetCityName(QString cityName);
     void SetTime(QDateTime t);
     void SetVibration(QString v);
     void SendNotify(unsigned int id, QString appName, QString icon, QString body, QString summary);
+    void SetWeatherLocation(const QString lat, const QString lng);
 
 private slots:
     void onTimeServiceReady();
     void onNotifyServiceReady();
     void onScreenshotServiceReady();
     void onWeatherServiceReady();
     void onDisconnected();
+    void onReplyFinished();
```

Comment:
Given that this class does a lot, it would be helpful to name this into something more specific, maybe onOwmReplyFinished?

Reply:
or maybe you could use a lambda and inline the slot in owmRequest. I think that's what the cool kids would do these days